lib/activerecord/cte/core_ext.rb

```diff
@@ -4,8 +4,31 @@ module ActiveRecord
   module Querying
     delegate :with, to: :all
   end
+  
+  module WithMerger
+    def normal_values
+      super + %i[with]
+    end
```

Comment:
Not sure why `normal_values` method is needed?

Reply:
@vlado my observation was that [`Merger` loops through a list of values](https://github.com/rails/rails/blob/v6.0.2.1/activerecord/lib/active_record/relation/merger.rb#L61) and this would cause it to implicitly follow the same process for `with`.

Reply:
Got it now. I was looking at the main branch https://github.com/rails/rails/blob/main/activerecord/lib/active_record/relation/merger.rb that does not use `normal_values`. I think this is fine for this PR then and we can change / improve when adding 6.1 compatibility.